src/number_theory/cyclotomic/discriminant.lean

```diff
@@ -20,15 +20,45 @@ We compute the discriminant of a `p`-th cyclotomic extension.
 -/
 
 universes u v
-variables {p : ℕ+} (k : ℕ) {K : Type u} {L : Type v} {ζ : L} [field K] [field L]
-variables [algebra K L] [ne_zero ((p : ℕ) : K)]
 
-open algebra polynomial nat is_primitive_root
-
-namespace is_cyclotomic_extension
+open algebra polynomial nat is_primitive_root power_basis
 
 local attribute [instance] is_cyclotomic_extension.finite_dimensional
 local attribute [instance] is_cyclotomic_extension.is_galois
+local attribute [instance] is_cyclotomic_extension.number_field
+
+namespace is_primitive_root
+
+open_locale polynomial
+
+variables {n : ℕ+} {K : Type u} [field K] [char_zero K] {ζ : K}
+variables [is_cyclotomic_extension {n} ℚ K]
+
+/-- The discriminant of the power basis given by a primitive root of unity `ζ` is the same as the
+discriminant of the power basis given by `ζ - 1`. -/
+lemma discr_zeta_eq_discr_zeta_sub_one (hζ : is_primitive_root ζ n) :
+  discr ℚ (hζ.power_basis ℚ).basis = discr ℚ (hζ.sub_one_power_basis ℚ).basis :=
+begin
+  have H₁ : (aeval (hζ.power_basis ℚ).gen) (X - 1 : ℤ[X]) = (hζ.sub_one_power_basis ℚ).gen :=
```

Comment:
Both of these power bases are a `ℤ`-basis for the ring of integers of `K`, right? Does Lean know this. Because then it should be a 1-liner that the discriminants are equal.
I think this is very close to the proof that you are giving, but it would be nice if this doesn't need 12 lines.